test/junit/scala/tools/nsc/backend/jvm/BTypesTest.scala

```diff
@@ -221,4 +222,30 @@ class BTypesTest extends BytecodeTesting {
   def maxTypeTest() {
 
   }
+
+  @Test
+  def maxValueTypeATest(): Unit = duringBackend {
+    assertEquals(LONG, LONG.maxValueType(BYTE))
+    assertEquals(LONG, LONG.maxValueType(SHORT))
+    assertEquals(LONG, LONG.maxValueType(CHAR))
+    assertEquals(LONG, LONG.maxValueType(INT))
+    assertEquals(LONG, LONG.maxValueType(LONG))
+    assertEquals(FLOAT, LONG.maxValueType(FLOAT))
+    assertEquals(DOUBLE, LONG.maxValueType(DOUBLE))
+
+    assertUncomparable(LONG, UNIT)
```

Comment:
You could use the `assertThrows` utility here: https://github.com/scala/scala/blob/2.13.x/src/testkit/scala/tools/testkit/AssertUtil.scala#L117

Reply:
I've rewritten `assertUncomparable` to use `AssertUtil.assertThrows` as suggested.

I've also tried to inline it but I think there is too much duplicated code that way. Let me know if you still want it to be inlined.

Reply:
Looks good, thank you! The advantage of `assertThrows` is that the test fails if the code doesn't throw the expected exception.

Reply:
Right, I forgot to guard against that case since I didn't expect the tests to fail. That's one reason why tests should be added first if possible.